### ISSUE TITLE
Increase data retention to 30 days

### ISSUE THREAD

**Issue by mdraijer** [TwiN/gatus#128]
For the uptime badges (1h, 24h, 7d), is it possible to also support a month or 30d? Maybe even 1y?

**Comment by sahil87** [comment]
Would love this feature

**Comment by TwiN** [comment]
Closed in favor of #242

**Comment by TwiN** [comment]
C

**Comment by mdraijer** [comment]
Do you have any idea about a timeline for this improvement?

**Comment by mdraijer** [comment]
By the way, @TwiN regarding your change of the title of this issue: I see that sometimes the data _is_ retained for at least 30 days, sometimes a lot longer. But in some cases a lot shorter. Probably depending on the 'MaximumNumberOfResults' and 'MaximumNumberOfEvents'? (couldn't find that in the documentation, didn't search the code yet)

So I assume that you meant to say "guarantee retention to 30 days; may be a lot longer if few events happen"?

**Comment by TwiN** [comment]
@mdraijer yeah since people can set their own interval, and these interval can be extremely short (1ms), it'd be a bad idea to save for the past 30 days. What I'll do instead is make MaximumNumberOfResults configurable.

**Comment by mdraijer** [comment]
Agreed, would be sufficient and a neat solution to have that value configurable.
Would that also make it possible to get badges for 30 days / 1 month?

**Comment by PurseChicken** [comment]
I would like to mention that this would be extremely desired from our team as well. It would be great to get 30d, 60d, 90d and 180d intervals.

**Comment by xeruf** [comment]
Yes, would love as well - we had a longer outage and an extended interval could have helped us to trace back the issue more swiftly!

**Comment by TwiN** [comment]
FYI #836 will close this. I know some of you want longer than 30d, but at least it's a significant step up from the previous maximum of 7d.

I haven't had the opportunity to completely test the feature, so when #836 is merged and new latest images are pushed, I'd appreciate if some of you could give it a test (of course, I'm aware it'll take some time before the data populates, but the more testers, the faster I can release)

Thanks!
